ui-tests/src/test/java/org/openmrs/reference/EndVisitTest.java

```diff
@@ -33,20 +32,20 @@ public void setUp() throws Exception {
     }
 
     @Test
-    @Ignore
     @Category(BuildTests.class)
-    public void EndVisitTest() throws Exception {
+    public void endVisitTest() {
         FindPatientPage findPatientPage = homePage.goToFindPatientRecord();
         findPatientPage.enterPatient(patient.identifier);
+        findPatientPage.waitForPageToLoad();
         ClinicianFacingPatientDashboardPage clinicianFacingPatientDashboardPage = findPatientPage.clickOnFirstPatient();
         PatientVisitsDashboardPage patientVisitsDashboardPage = clinicianFacingPatientDashboardPage.goToRecentVisits();
-        patientVisitsDashboardPage = patientVisitsDashboardPage.endVisit();
+        patientVisitsDashboardPage.endVisit();
+        patientVisitsDashboardPage.waitForPageToLoad();
```

Comment:
`Waiting the page to load` ensures that the `endVist()` operation is effective otherwise by the time `assertNull()` applies its logic to affirm null active visit unfortunately finds it still active due to delay of the `endVisit()` operation on the visit.